### PR TITLE
Make data csr and C_CONTIGUOUS if sparse or  dense np respectively

### LINKED ISSUE #787: Ensure protein and X are "C" order if dense
I already did this if using a layer, but we should always check whenever it's a dense numpy array.

In the case of proteins, if it's a dataframe we may need to overwrite the obsm with a new dataframe.

### PR DESCRIPTION
**Pull request by galenxing**
Fixes #787

### CODE REVIEW

File: tests/dataset/test_anndata.py [hunk 1]
```diff
@@ -99,6 +100,41 @@ def test_transfer_anndata_setup():
...
+    old_X = adata.X
+    old_pro = adata.obsm["protein_expression"]
+    old_obs = adata.obs
+    adata.X = np.asfortranarray(old_X)
+    adata.obsm["protein_expression"] = np.asfortranarray(old_pro)
+    assert adata.X.flags["C_CONTIGUOUS"] is False
+    assert adata.obsm["protein_expression"].flags["C_CONTIGUOUS"] is False
+
+    setup_anndata(adata, protein_expression_obsm_key="protein_expression")
+    assert adata.X.flags["C_CONTIGUOUS"] is True
```
Comment (adamgayoso) [inline tests/dataset/test_anndata.py:116]: let's make sure using get from registry gives the same results

File: scvi/data/_anndata.py [hunk 2]
```diff
@@ -240,12 +241,77 @@ def setup_anndata(
...
+    adata
+        anndata to check
+    data_registry
+        data registry of anndata
+    """
+    for k in data_registry.keys():
+        data = get_from_registry(adata, k)
+        if isspmatrix(data):
+            data = data.tocsr()
+        elif isinstance(data, np.ndarray):
```
Comment (adamgayoso) [inline scvi/data/_anndata.py:310]: I think we are missing the case if it's a DataFrame here, like for the proteins

File: scvi/data/_anndata.py [hunk 3]
```diff
@@ -240,12 +241,77 @@ def setup_anndata(
...
+            attribute.loc[:, attr_key] = data
+        else:
+            attribute[attr_key] = data
+        setattr(tmp_adata, attr_name, attribute)
+
+    if use_raw is True and attr_name in ["X", "var"]:
+        adata.raw = tmp_adata
+
+
+def _verify_and_correct_data_format(adata, data_registry):
```
Comment (adamgayoso) [inline scvi/data/_anndata.py:293]: I feel like we should only care about X and proteins for now, maybe a parameter here for which keys to check? If none, check all?

File: tests/dataset/test_anndata.py (hunk 1 above)
Comment (adamgayoso) [reply tests/dataset/test_anndata.py:116]: I think I had noticed a weird issue where you set adata.X of a registered object and get from registry still references the old object

File: scvi/data/_anndata.py [hunk 4]
```diff
@@ -240,12 +241,92 @@ def setup_anndata(
...
+        anndata to check
+    data_registry
+        data registry of anndata
+    """
+    keys_to_check = [_CONSTANTS.X_KEY, _CONSTANTS.PROTEIN_EXP_KEY]
+    keys = [key for key in keys_to_check if key in data_registry.keys()]
+
+    for k in keys:
+        data = get_from_registry(adata, k)
+        if isspmatrix(data) and (data.getformat() == "csc"):
```
Comment (adamgayoso) [inline scvi/data/_anndata.py:311]:
```suggestion
        if isspmatrix(data) and (data.getformat() != "csr"):
```

File: scvi/data/_anndata.py [hunk 5]
```diff
@@ -240,12 +241,92 @@ def setup_anndata(
...
+    Will iterate through all the keys of data_registry.
+
+    Parameters
+    ----------
+    adata
+        anndata to check
+    data_registry
+        data registry of anndata
+    """
+    keys_to_check = [_CONSTANTS.X_KEY, _CONSTANTS.PROTEIN_EXP_KEY]
```
Comment (adamgayoso) [inline scvi/data/_anndata.py:306]: what happens if protein exp isn't registered?

File: tests/dataset/test_anndata.py (hunk 1 above)
Comment (adamgayoso) [reply tests/dataset/test_anndata.py:116]: I would still add one line to the test to make sure.

File: scvi/data/_anndata.py (hunk 5 above)
Comment (galenxing) [reply scvi/data/_anndata.py:306]: The next line will check if its in data_registry

Comment (adamgayoso) [reply scvi/data/_anndata.py:306]: ah, didn't see

File: scvi/data/_anndata.py [hunk 6]
```diff
@@ -240,12 +241,98 @@ def setup_anndata(
...
+            data = data.tocsr()
+            _set_data_in_registry(adata, data, k)
+        elif isinstance(data, np.ndarray) and (data.flags["C_CONTIGUOUS"] is False):
+            logger.debug(
+                "{} is not C_CONTIGUOUS. Overwriting to C_CONTIGUOUS.".format(k)
+            )
+            data = np.asarray(data, order="C")
+            _set_data_in_registry(adata, data, k)
+        elif isinstance(data, pd.DataFrame) and (
+            data.values.flags["C_CONTIGUOUS"] is False
```
Comment (adamgayoso) [inline scvi/data/_anndata.py:322]: values or to_numpy?

Comment (galenxing) [reply scvi/data/_anndata.py]: seems like values is deprecated. just changed to to_numpy()

**Review by adamgayoso — APPROVED** [review]

**Pull request merged**